src/stream/cycle.rs

```diff
@@ -40,21 +71,21 @@ impl<T: Copy> Stream for Cycle<T> {
 /// use async_std::prelude::*;
 /// use async_std::stream;
 ///
-/// let mut s = stream::cycle(vec![1,2,3]);
+/// let mut s = stream::cycle(stream::once(7));
 ///
-/// assert_eq!(s.next().await, Some(1));
-/// assert_eq!(s.next().await, Some(2));
-/// assert_eq!(s.next().await, Some(3));
-/// assert_eq!(s.next().await, Some(1));
-/// assert_eq!(s.next().await, Some(2));
+/// assert_eq!(s.next().await, Some(7));
+/// assert_eq!(s.next().await, Some(7));
+/// assert_eq!(s.next().await, Some(7));
+/// assert_eq!(s.next().await, Some(7));
+/// assert_eq!(s.next().await, Some(7));
 /// #
 /// # })
 /// ```
-pub fn cycle<T: Copy>(source: Vec<T>) -> impl Stream<Item = T> {
-    let len = source.len();
+pub fn cycle<S: Stream<Item = T>, T: Clone>(source: S) -> impl Stream<Item = S::Item> {
```

Comment:
This should probably be a method on `Stream`, and not a free function. Similar to [`Iterator::cycle`](https://doc.rust-lang.org/core/iter/trait.Iterator.html#method.cycle)

Reply:
Done 🎉